analysisservice/src/test/java/nu/yona/server/analysis/service/AnalysisEngineServiceTests.java

```diff
@@ -624,12 +622,91 @@ public void analyze_crossDayAppActivity_twoDayActivitiesCreated()
 		Activity nextPart = todaysDayActivity.getLastActivity();
 
 		assertThat(firstPart.getStartTime(), equalTo(startTime.toLocalDateTime()));
-		ZonedDateTime lastMidnight = ZonedDateTime.now(userAnonZoneId).truncatedTo(ChronoUnit.DAYS);
-		assertThat(firstPart.getEndTime(), equalTo(lastMidnight.minusSeconds(1).toLocalDateTime()));
+		ZonedDateTime lastMidnight = now().truncatedTo(ChronoUnit.DAYS);
+		assertThat(firstPart.getEndTime(), equalTo(lastMidnight.toLocalDateTime()));
 		assertThat(nextPart.getStartTime(), equalTo(lastMidnight.toLocalDateTime()));
 		assertThat(nextPart.getEndTime(), equalTo(endTime.toLocalDateTime()));
 	}
 
+	@Test
+	public void analyze_appActivityAfterNetworkActivityWithinConflictInterval_newActivityRecordCreatedButNoNewGoalConflictMessagesCreated()
+	{
+		ZonedDateTime now = now();
+		DayActivity existingDayActivity = mockExistingActivity(gamblingGoal, now());
+
+		service.analyze(userAnonId, createSingleAppActivity("Lotto App", now.minusMinutes(4), now.minusMinutes(2)));
+
+		verifyNoGoalConflictMessagesCreated();
+		List<Activity> activities = existingDayActivity.getActivities();
+		assertThat(activities.size(), equalTo(2));
+		assertThat(activities.get(1).getApp(), equalTo(Optional.of("Lotto App")));
+		assertThat(activities.get(1).getDurationMinutes(), equalTo(2));
+	}
+
+	@Test
+	public void analyze_networkActivityAfterAppActivityWithinConflictInterval_newActivityRecordCreatedButNoNewGoalConflictMessagesCreated()
+	{
+		ZonedDateTime now = now();
+		DayActivity existingDayActivity = mockExistingActivity(gamblingGoal, now.minusMinutes(10), now.minusMinutes(5),
+				"Lotto App");
+
+		service.analyze(userAnonId, createNetworkActivityForCategories("lotto"));
+
+		verifyNoGoalConflictMessagesCreated();
+		List<Activity> activities = existingDayActivity.getActivities();
+		assertThat(activities.size(), equalTo(2));
+		assertThat(activities.get(1).getApp(), equalTo(Optional.empty()));
+		assertThat(activities.get(1).getDurationMinutes(), equalTo(1));
+	}
+
+	@Test
+	public void analyze_appActivityDifferentAppWithinConflictInterval_newActivityRecordCreatedButNoNewGoalConflictMessagesCreated()
+	{
+		ZonedDateTime now = now();
+		DayActivity existingDayActivity = mockExistingActivity(gamblingGoal, now.minusMinutes(10), now.minusMinutes(5),
+				"Poker App");
+
+		service.analyze(userAnonId, createSingleAppActivity("Lotto App", now.minusMinutes(4), now.minusMinutes(2)));
+
+		verifyNoGoalConflictMessagesCreated();
+		List<Activity> activities = existingDayActivity.getActivities();
+		assertThat(activities.size(), equalTo(2));
+		assertThat(activities.get(1).getApp(), equalTo(Optional.of("Lotto App")));
+		assertThat(activities.get(1).getDurationMinutes(), equalTo(2));
+	}
+
+	@Test
+	public void analyze_appActivitySameAppWithinConflictIntervalContinuous_activityRecordMergedAndNoNewGoalConflictMessageCreated()
```

Comment:
The funny thing is that it's actually "onderbroken": there are two app activities that are adjacent with no gap.